src/templates/Practice4.tsx

```diff
@@ -0,0 +1,132 @@
+import { useForm} from 'react-hook-form'
+import { object, string,InferType } from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useState} from 'react';
+import Button from '../components/Button'
+
+  const validationSchema = object().shape({
+    postalCode: string()
+      .required('必須項目です')
+      .matches(/^\d{7}$/, '7桁の数字を入力してください'),
+    prefecture: string().required('必須項目です'),
+    city: string()
+      .required('必須項目です'),
+  }).required();
+
+type FormData = InferType<typeof validationSchema>;
+
+const Practice4: React.FC = () => {
+  const [canSubmit, setCanSubmit] = useState<Boolean>(true);
+  const [canSearch, setCanSearch] = useState<Boolean>(true);
```

Comment:
According to our [Typescript conventions](https://pj100.esa.io/posts/16606#boolean%E3%81%AE%E5%91%BD%E5%90%8D%E3%81%AB%E3%81%AFis%E3%80%87%E3%80%87%E3%82%82%E3%81%97%E3%81%8F%E3%81%AFhas%E3%80%87%E3%80%87%E3%82%92%E4%BD%BF%E7%94%A8%E3%81%99%E3%82%8B%E3%80%82), Boolean state variables should include the word `is` or `has` to make it easily recognizable as a boolean by other team members.

You might want to consider renaming them to `isSubmittable` and `isSearchable` instead.

Also, instead of `Boolean` (which is a constructor), you should use the primitive type `boolean` (lowercase b) instead.

Reply:
Made the changes !